src/router.js

```diff
@@ -439,7 +455,7 @@ export class Router extends Resolver {
           fireRouterEvent('location-changed', {router: this, location: this.location});
 
           if (shouldUpdateHistory) {
-            this.__updateBrowserHistory(context.pathname, context.redirectFrom);
+            this.__updateBrowserHistory(context, context.redirectFrom);
```

Comment:
If the entire `context` is passed as a parameter, what's the main reason to pass the `context.redirectFrom` property as a separate second parameter?

Reply:
Separation of concerns. This argument is named `replace`, it’s a boolean flag on whether the new location replaces or is pushed after the old one. The method is not supposed to decide this, I think, but the caller should.